RecoPixelVertexing/PixelTriplets/plugins/CAConstants.h

```diff
@@ -0,0 +1,69 @@
+#ifndef RecoPixelVertexing_PixelTriplets_plugins_CAConstants_h
+#define RecoPixelVertexing_PixelTriplets_plugins_CAConstants_h
+
+#include <cstdint>
+
+#include <cuda_runtime.h>
+
+#include "CUDADataFormats/SiPixelCluster/interface/gpuClusteringConstants.h"
+#include "HeterogeneousCore/CUDAUtilities/interface/SimpleVector.h"
+#include "HeterogeneousCore/CUDAUtilities/interface/VecArray.h"
+#include "HeterogeneousCore/CUDAUtilities/interface/HistoContainer.h"
+
+// #define ONLY_PHICUT
+
+namespace CAConstants {
+
+  // constants
+#ifndef ONLY_PHICUT
+#ifdef GPU_SMALL_EVENTS
+  constexpr uint32_t maxNumberOfTuples() { return 3 * 1024; }
+#else
+  constexpr uint32_t maxNumberOfTuples() { return 24 * 1024; }
+#endif
+#else
+  constexpr uint32_t maxNumberOfTuples() { return 48 * 1024; }
+#endif
+  constexpr uint32_t maxNumberOfQuadruplets() { return maxNumberOfTuples(); }
+#ifndef ONLY_PHICUT
+#ifndef GPU_SMALL_EVENTS
+  constexpr uint32_t maxNumberOfDoublets() { return 512 * 1024; }
+  constexpr uint32_t maxCellsPerHit() { return 128; }
+#else
+  constexpr uint32_t maxNumberOfDoublets() { return 128 * 1024; }
+  constexpr uint32_t maxCellsPerHit() { return 128 / 2; }
+#endif
+#else
+  constexpr uint32_t maxNumberOfDoublets() { return 2 * 1024 * 1024; }
+  constexpr uint32_t maxCellsPerHit() { return 8 * 128; }
+#endif
+  constexpr uint32_t maxNumOfActiveDoublets() { return maxNumberOfDoublets() / 8; }
+
+  constexpr uint32_t maxNumberOfLayerPairs() { return 20; }
+  constexpr uint32_t maxNumberOfLayers() { return 10; }
```

Comment:
I'm afraid this will not work properly for Phase2 geometry. am I wrong ?
the same applies to the previous line